gobblin-compaction/src/main/java/gobblin/compaction/Dataset.java

```diff
@@ -118,13 +112,13 @@ public Dataset build() {
   private Path inputPath;
   private Path inputLatePath;
   private State jobProps;
+  private Optional<State> extraDatasetProps;
```

Comment:
Can remove `extraDatasetProps` and reuse  `jobProps`?
You can add a method `Dataset.setJobProp()` that sets a single property.